### ISSUE TITLE
FSQ Missing mystery tanh from Stable Codec DitheredFSQ

### ISSUE THREAD

**Issue by zaptrem** [lucidrains/vector-quantize-pytorch#193]
This [commit](https://github.com/lucidrains/vector-quantize-pytorch/commit/09c33f3f6c6632c0345db0acf7f2fecbd6e34e8e) from @lucasnewman added some of the major features from Stable Codec's improved FSQ. However, it's missing this tanh application [here](https://github.com/Stability-AI/stable-codec/blob/853554419944dbec11c39a29cf803e492aade836/stable_codec/fsq.py#L65). Adding it significantly harms (not improves) reconstruction loss during early training at least, but maybe it serves some important purpose? There are a number of other small numerical differences between the implementations as well such as the order in which quantization dropout is applied and a few scaling factors, but none have a significant effect on reconstruction loss compared to this change.

**Comment by zaptrem** [comment]
Never mind, it's [here](https://github.com/lucidrains/vector-quantize-pytorch/blob/09c33f3f6c6632c0345db0acf7f2fecbd6e34e8e/vector_quantize_pytorch/finite_scalar_quantization.py#L129).
